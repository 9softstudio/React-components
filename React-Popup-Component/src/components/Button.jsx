import React, { Component } from 'react'
import PropTypes from 'prop-types';

export default class Button extends Component {
    render() {
        const { text, className, onButtonClick } = this.props;
        return <button className={className} onClick={onButtonClick}>{text}</button>
    }
}

Button.propTypes = {
    text: PropTypes.string,
    className: PropTypes.string,
    onButtonClick: PropTypes.func.isRequired
}
