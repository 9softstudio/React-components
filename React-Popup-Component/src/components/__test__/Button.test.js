import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';

describe('<Button />', () => {
    test('renders correctly', () => {
        const component = renderer
                .create(<Button text="fake-text" className="fake-class-name" onButtonClick={jest.fn()} />);
                
        expect(component.toJSON()).toMatchSnapshot();
    });
});
