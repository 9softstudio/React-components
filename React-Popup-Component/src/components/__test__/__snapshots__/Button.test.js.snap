// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="fake-class-name"
  onClick={[MockFunction]}
>
  fake-text
</button>
`;
